enhancements/authentication/improved-login-workflow.md

```diff
@@ -0,0 +1,242 @@
+---
+title: oc-authorization-grant-login
+authors:
+  - "@arjunrn"
+reviewers:
+  - "@stlaz"
+  - "@s-urbaniak"
+  - "@soltysh"
+approvers:
+  - "@soltysh"
+  - "@mfojtik"
+creation-date: 2021-11-08
+last-updated: 2021-11-08
+status: provisional
+---
+
+# oc Authorization Grant Login
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes a new workflow for the `oc login` command which requires fewer logins and no copy-pasting of
+sensitive credentials. The new workflow is based on implementing some functionality in
+the [oauth-server](https://github.com/openshift/oauth-server) and [oc](https://github.com/openshift/oc) as specified
+in [RFC-8252](https://datatracker.ietf.org/doc/html/rfc8252).
+
+## Motivation
+
+The current CLI login process involves multiple interactive steps where the user has to enter credentials multiple times
+in the browser. This process can be simplified so that the `oc` tool can be used to login directly without having to
+first login to the cluster console, the OAuth server and finally copying the login command. The `oc` tool can be
+modified to act as a public client to the Openshift OAuth server and received an access token
+through [Authorization Code Grant](https://datatracker.ietf.org/doc/html/rfc6749#section-1.3) flow
+and [PKCE](https://datatracker.ietf.org/doc/html/rfc7636).
+
+### Goals
+
+Provide a single-step login in `oc` for identity providers that don't post HTTP challenges.
```

Comment:
For me the phrasing "single-step" confused me completely. As far as I can tell there are more steps for OAuth2 for Native Apps:

- Native App triggers flow (oc login)
- Browser opens
- if not logged in, user needs to enter credentials
- User needs to approve the resources to be shared
- redirect

Reply:
What I meant to say in there was - provide a way for users to login such that they only have to authenticate once. Currently users have to open the console, then click on the login command button which opens the token page where they have to login all over again before they copy the command and paste it into the console. The new approach instead means that if they user wants to login again to a cluster they have previous logged into again all they have to do is run `oc login --browser` and authenticate if required. The `oc` client will then be reconfigured automatically with the new token. 

If you can think of a good way of describing this in the goal I'm don't mind changing it. I understand it's not the best description.